xgi/drawing/draw.py

```diff
@@ -1947,24 +1936,43 @@ def draw_undirected_dyads(
     else:
         edge_ids = list(H.edges.filterby("order", max_order, "leq"))
 
+    node_pos, edge_pos = pos
+
+    dyads = to_bipartite_edgelist(H)
+    dyad_pos = np.asarray(
+        [(node_pos[e[0]], edge_pos[e[1]]) for e in dyads if e[1] in edge_ids]
+    )
+
     dyad_lw = _draw_arg_to_arr(dyad_lw)
 
     if rescale_sizes and isinstance(dyad_lw, np.ndarray):
         dyad_lw = _interp_draw_arg(
             dyad_lw, settings["min_dyad_lw"], settings["max_dyad_lw"]
         )
 
-    # set default colors
-    if dyad_color is None:
-        dyad_color = H.edges.size
-
     # parse colors
     dyad_color, dyads_c_mapped = _parse_color_arg(dyad_color, H.edges)
 
+    if isinstance(dyad_lw, np.ndarray):
+        dyad_lw = np.array(
+            list(
+                chaini([lw] * int(s) for s, lw in zip(H.edges.size.aslist(), dyad_lw)),
+                dtype=float,
+            )
+        )
+
+    if isinstance(dyad_color, np.ndarray):
+        dyad_color = np.array(
+            list(
+                chaini(
+                    [dc] * int(s) for s, dc in zip(H.edges.size.aslist(), dyad_color)
+                ),
+            )
+        )
+
```

Comment:
what does this do? Multiplies the number of elements of linewidth by the size of each edge?

Reply:
This allows you to specify a color for each edge ID and automatically apply it to all associated bipartite edges instead of specifying a color for all bipartite edges.

Reply:
I added a comment in the code explaining what these lines do.